examples/reg_debugger.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+# Copyright (c) PLUMgrid, Inc.
+# Licensed under the Apache License, Version 2.0 (the "License")
+#
+# This example shows how a BPF program can single step through the calculations
+# in a single function. The use case in this scenario was to see the
+# step-by-step math in an ipv6 multipath route selection to debug why the
+# distribution was uneven.
+# The address offsets are unstable, and should be used with the kernel
+# assembler output side by side.
+
+from bpf import BPF
+
+def reg_printer(name, reg, wrap=""):
+    fn = """
+int {0}(struct pt_regs *ctx) {{
+    return bpf_trace_printk("{1}=0x%llx\\n", {2}(ctx->{1}));
+}}
+"""
+    return fn.format(name, reg, wrap)
+
+regs = ["di", "si", "dx", "cx", "ax", "bx", "bp"]
+regs += ["r%d" % i for i in range(8, 15)]
+
+
+prog = """
+#include <linux/ptrace.h>
+"""
+for r in regs:
+    prog += reg_printer("print_%s" % r, r)
+    prog += reg_printer("print_%s_net" % r, r, "bpf_ntohll")
```

Comment:
I'm ok leaving this code out of the branch. I just wanted to get it out of my private workspace and somewhere visible. We can close the pull request and refer to it as needed or resurrect when a valid use case is found.